File: Count of Subarrays - GFG/count-of-subarrays.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

#define ll long long


// } Driver Code Ends
//User function template for C++
class Solution{
public:
	#define ll long long

	ll countSubarray(int arr[], int n, int k) {
	    
        ll ans = 0,count = 0;
        for(int i=0;i<n;i++){
            if(arr[i] > k){
                ans = ans + (count*(count+1)/2);
                count=0;
            } else {
                count++;
            }
        }
        ans = ans+ (count*(count+1)/2);
        ll N = n;
        ll total = N*(N+1)/2;
        return total-ans;
	}

};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, k, i;
        cin >> n >> k;
        int arr[n];
        for (i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.countSubarray(arr, n, k);
        cout << ans << "\n";
    }
    return 0;
}
// } Driver Code Ends

File: Count of Subarrays - GFG/count-of-subarrays.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            String[] str = br.readLine().trim().split(" ");
            int n = Integer.parseInt(str[0]);
            int k = Integer.parseInt(str[1]);
            str = br.readLine().trim().split(" ");
            int[] arr = new int[n];
            for (int i = 0; i < n; i++) {
                arr[i] = Integer.parseInt(str[i]);
            }

            long ans = new Solution().countSubarray(arr, n, k);

            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {

    long countSubarray(int arr[], int n, int k) {
        long N = n;
        long total = N*(N+1)/2;
        
        long ans = 0l,count = 0l;
        for(int i=0;i<n;i++){
            if(arr[i] > k){
                ans = ans + (count*(count+1)/2);
                count=0;
            } else {
                count++;
            }
        }
        ans = ans + (count*(count+1)/2);
        return Math.abs(total-ans);
    }
}